crates/ra_lsp_server/src/main_loop.rs

```diff
@@ -337,18 +337,42 @@ fn loop_turn(
             loop_state.in_flight_libraries -= 1;
         }
         Event::CheckWatcher(task) => match task {
```

Comment:
Let's move the contents of this `match` to a separate function, and maybe pass `&mut State` and `&Vfs` in, to avoid locking locks repeatedly?

It also seems like at least notificaiton-sending bit can be extracted as well?

That is:
* let's move more code out of the main loop itself, to make the overall structure of the loop clearer
* let's try to minimize the amount of lock locking/unlocking, to make it clearer how the state changes
* let's try to reduce code duplicaiton